lib/webdrivers/mswebdriver.rb

```diff
@@ -23,32 +28,30 @@ def file_name
         "MicrosoftWebDriver.exe"
       end
 
-      def download_url(_version = nil)
+      def downloads
         raise StandardError, "Can not reach site" unless site_available?
 
-        case current.to_i
-        when 10240
-          Webdrivers.logger.debug "Attempting to Download Build for 10240"
-          "https://download.microsoft.com/download/8/D/0/8D0D08CF-790D-4586-B726-C6469A9ED49C/MicrosoftWebDriver.msi"
-        when 10586
-          Webdrivers.logger.debug "Attempting to Download Build for 10586"
-          "https://download.microsoft.com/download/C/0/7/C07EBF21-5305-4EC8-83B1-A6FCC8F93F45/MicrosoftWebDriver.msi"
-        when 14393
-          Webdrivers.logger.debug "Attempting to Download Build for 14393"
-          "https://download.microsoft.com/download/3/2/D/32D3E464-F2EF-490F-841B-05D53C848D15/MicrosoftWebDriver.exe"
-        when 15063
-          Webdrivers.logger.debug "Attempting to Download Build for 15063"
-          "https://download.microsoft.com/download/3/4/2/342316D7-EBE0-4F10-ABA2-AE8E0CDF36DD/MicrosoftWebDriver.exe"
-        else
-          Webdrivers.logger.debug "Attempting to Download Latest Insider's Version"
-          "https://download.microsoft.com/download/1/4/1/14156DA0-D40F-460A-B14D-1B264CA081A5/MicrosoftWebDriver.exe"
-        end
+        available = {} # version => download_url
+        doc       = Nokogiri::HTML(get(download_page))
+        doc.xpath("//li[@class='driver-download']/a").map { |link|
+          next if link.text == 'Insiders' # Ignore Insiders release
+          available[link.text.scan(/\d+/).first.to_i] = link['href']
+        }
+        available
       end
 
       def base_url
         'https://www.microsoft.com/en-us/download'
       end
 
+       # Using this as base_url yields Net::HTTPRedirection with a partial location, which fails
+       # on the second Common#get call in lib/webdrivers/common.rb:77.
+       # Using existing base_url to get past that hurdle until we discuss a better way to handle
+       # the HTTP status.
+      def download_page
+        "https://developer.microsoft.com/en-us/microsoft-edge/tools/webdriver/"
```

Comment:
Can this page be the base_url? Not sure what the distinction is, even with your comment.

Reply:
Huh... so I was running into issues when with HTTP redirection when using the `download_page` URL. I wonder if it was because I was behind a proxy at work when I worked on this, but now that I am at home I am not experiencing it. I will have to test this out on Monday. For now, I have removed `download_page` and update `base_url`.